plugins/nf-ga4gh/src/main/nextflow/ga4gh/tes/executor/TesExecutor.groovy

```diff
@@ -44,11 +48,18 @@ class TesExecutor extends Executor implements ExtensionPoint {
 
     private TaskServiceApi client
 
+    /**
+     * A path accessible to TES where executable scripts need to be uploaded
+     */
+    private Path remoteBinDir = null
+
     @Override
     protected void register() {
-        if( session.binDir && !session.binDir.empty() ) {
+        if( session.binDir && !session.binDir.empty() && !session.workDir ) {
             session.abort()
-            throw new AbortOperationException("ERROR: TES executor does not allow the use of custom scripts in the `bin` folder")
+            throw new AbortOperationException("ERROR: TES executor does not allow the use of custom scripts in the `bin` folder without a workdir")
```

Comment:
You can remove this check entirely, the work directory is always defined.

Reply:
Or if you're saying that it only works with local/shared filesystem, you can check that with `session.workDir.fileSystem == FileSystems.default`. That condition will be false when the work directory is object storage like S3.